airqo-platform/AirQo-frontend
# [NetManager] Replace device modals with actual pages -- part - 1

#### Summary of Changes (What does this PR do?)
- Update device and location registries to use redux store (to avoid unnecessary api queries)
- Added more redux state ( maintenance logs, components and main Alert)
- Completed Maintenance logs and components pages.
- Added provisional photo page.

#### Is this change ready to hit production in its current state?

#### Status of maturity (all need to be checked before merging):


#### How should this be manually tested?
* Pull and checkout to this [branch](https://github.com/airqo-platform/AirQo-frontend/tree/replace-device-modals-PLAT-264) locally
* Install node requirements `npm install`
* Start the application `npm run stage-mac` (on mac) or `npm run stage-pc` (on windows platform)
* On loading devices and locations, navigating to another page then back should not trigger a reload
* Navigate to device view, components and maintenance logs pages are fully functional

#### What are the relevant tickets?
- [PLAT-264](https://airqoteam.atlassian.net/browse/PLAT-264)

#### Screenshots (optional)


- Components page

![image](https://user-images.githubusercontent.com/39955305/97406616-8cf27780-190a-11eb-919c-f4233e724abf.png)


- App main alert

![image](https://user-images.githubusercontent.com/39955305/97406688-abf10980-190a-11eb-9775-8f624111d4dd.png)
